### PR TITLE
chore: Otel upgrade to 0.102.1-sumo-0

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Upgrading to this version should address two things
- Fix known CVEs in otel
- Address an ongoing customer issue with multiline log parsing

### CODE REVIEW

**Review by echlebek — APPROVED** [review]

**Comment by echlebek** [comment]
Not sure what's going on with the integration tests as there is no useful output there. :thinking:

**Comment by rnishtala-sumo** [comment]
yes need to understand why upgrading otel is breaking the integration tests.

**Comment by sumo-drosiek** [comment]
Integration tests has not been run. The failing job is the summary of integration tests

File: tests/integration/helm_otlp_test.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ func Test_Helm_OTLP(t *testing.T) {
 
 	expectedMetrics := internal.DefaultExpectedMetrics
 	// we have tracing enabled, so check tracing-specific metrics
-	expectedMetrics = append(expectedMetrics, internal.TracingOtelcolMetrics...)
+	expectedMetrics = append(expectedMetrics, internal.TracingOtelcolMetrics..., internal.FileConsumerOtelColMetrics...)
```
Comment (sumo-drosiek) [inline tests/integration/helm_otlp_test.go:16]: In https://github.com/SumoLogic/sumologic-kubernetes-collection/actions/runs/10107195658/job/27950661632 `Set test names`

```
Run echo "test_names_allversions=$(make list-tests-allversions | jq -R . | jq -cs .)" >> $GITHUB_OUTPUT
  echo "test_names_allversions=$(make list-tests-allversions | jq -R . | jq -cs .)" >> $GITHUB_OUTPUT
  echo "test_names_onlylatest=$(make list-tests-onlylatest | jq -R . | jq -cs .)" >> $GITHUB_OUTPUT
  shell: /usr/bin/bash -e {0}
  env:
    KIND_VERSION: 0.17.0
  
# github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration
Error: helm_ot_default_test.go:16:79: expected ')', found internal
# github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration
Error: helm_otlp_test.go:16:79: expected ')', found internal
```

**Pull request merged**